### DISCUSSION TITLE
InfiniteScroll 配合居于底部的TabBar 会一直触发loadMore 函数

### DISCUSSION THREAD

**Discussion by hsulei** [ant-design/ant-design-mobile#6787] [Q&A]
我使用基于底部的<TabBar> 组件, TabBar 永远固定在底部不懂，
```
<div className="w-full h-[99vh] flex flex-col">
			<div className="w-full flex-1 overflow-scroll">
				<Outlet />
			</div>
			<div className="w-full ">
				<TabBar activeKey={pathname} onChange={value => { setRouteActive(value) }} safeArea>
					<TabBar.Item key="11" icon={<AppOutline />} title="首页" />
				</TabBar >
			</div>
		</div>
```
类似这种模式，上半部分使用react-router， 进行路由，展示的时候 
```
const loadMore = async () => {

		const data = await pageXXX(page + 1)
		Toast.clear()

		setPage(page + 1)
		setDataSource(draft => {
			draft.push(...data.rows)
		})
		setHasMore(data.total > dataSource.length)
	}
<PullToRefresh
			onRefresh={handleRefresh}
		>
			<List>
				<AutoSizer disableHeight>
					{({ width }: { width: number }) => (
						<VirtualizedList
							rowCount={dataSource.length}
							rowRenderer={rowRenderer}
							width={width}
							height={window.innerHeight - 60}
							rowHeight={90}
							overscanRowCount={10}
						/>
					)}
				</AutoSizer>
			</List>

		</PullToRefresh>
		<InfiniteScroll loadMore={loadMore} hasMore={hasMore} />
```
这个loadMore函数无论咋样都会触发，知道hasMore为false ，是哪里写的有问题吗？

**Comment by Layouwen** [comment]
> 我使用基于底部的<TabBar> 组件, TabBar 永远固定在底部不懂，
> 
> ```
> 
> <div className="w-full h-[99vh] flex flex-col">
> 
> 			<div className="w-full flex-1 overflow-scroll">
> 
> 				<Outlet />
> 
> 			</div>
> 
> 			<div className="w-full ">
> 
> 				<TabBar activeKey={pathname} onChange={value => { setRouteActive(value) }} safeArea>
> 
> 					<TabBar.Item key="11" icon={<AppOutline />} title="首页" />
> 
> 				</TabBar >
> 
> 			</div>
> 
> 		</div>
> 
> ```
> 
> 类似这种模式，上半部分使用react-router， 进行路由，展示的时候 
> 
> ```
> 
> const loadMore = async () => {
> 
> 
> 
> 		const data = await pageXXX(page + 1)
> 
> 		Toast.clear()
> 
> 
> 
> 		setPage(page + 1)
> 
> 		setDataSource(draft => {
> 
> 			draft.push(...data.rows)
> 
> 		})
> 
> 		setHasMore(data.total > dataSource.length)
> 
> 	}
> 
> <PullToRefresh
> 
> 			onRefresh={handleRefresh}
> 
> 		>
> 
> 			<List>
> 
> 				<AutoSizer disableHeight>
> 
> 					{({ width }: { width: number }) => (
> 
> 						<VirtualizedList
> 
> 							rowCount={dataSource.length}
> 
> 							rowRenderer={rowRenderer}
> 
> 							width={width}
> 
> 							height={window.innerHeight - 60}
> 
> 							rowHeight={90}
> 
> 							overscanRowCount={10}
> 
> 						/>
> 
> 					)}
> 
> 				</AutoSizer>
> 
> 			</List>
> 
> 
> 
> 		</PullToRefresh>
> 
> 		<InfiniteScroll loadMore={loadMore} hasMore={hasMore} />
> 
> ```
> 
> 这个loadMore函数无论咋样都会触发，知道hasMore为false ，是哪里写的有问题吗？

提供个在线复现地址看看？😐
